synapse/handlers/room_member.py

```diff
@@ -210,24 +210,40 @@ async def _local_membership_update(
             _, stream_id = await self.store.get_event_ordering(duplicate.event_id)
             return duplicate.event_id, stream_id
 
-        stream_id = await self.event_creation_handler.handle_new_client_event(
-            requester, event, context, extra_users=[target], ratelimit=ratelimit,
-        )
-
         prev_state_ids = await context.get_prev_state_ids()
 
         prev_member_event_id = prev_state_ids.get((EventTypes.Member, user_id), None)
 
+        newly_joined = False
         if event.membership == Membership.JOIN:
-            # Only fire user_joined_room if the user has actually joined the
-            # room. Don't bother if the user is just changing their profile
-            # info.
             newly_joined = True
             if prev_member_event_id:
                 prev_member_event = await self.store.get_event(prev_member_event_id)
                 newly_joined = prev_member_event.membership != Membership.JOIN
+
+            # Only rate-limit if the user actually joined the room, otherwise we'll end
+            # up blocking profile updates.
             if newly_joined:
-                await self._user_joined_room(target, room_id)
+                time_now_s = self.clock.time()
+                (
+                    allowed,
+                    time_allowed,
+                ) = self._join_rate_limiter_local.can_requester_do_action(requester)
```

Comment:
This still doesn't fix https://github.com/matrix-org/synapse/issues/8146. If the user is in the ratelimit_override table and they join lots of rooms, they shouldn't get rate limited. We need to ensure that we check the table when calling can_requester_do_action.

Reply:
See https://github.com/matrix-org/synapse/issues/8146#issuecomment-679168222. The `ratelimit_override` table is only used for the ratelimiter defined by `rc_message` in the config; it doesn't apply to the other ratelimiters (join, login, register, etc.). It can possibly also be an issue, but AFAICT the main problem with #8146 is that `/join` gets ratelimited even if the user is already in the room.

Reply:
I think the missing information was in your second comment. Thanks for clearing that up.